src/array.jl

```diff
@@ -248,7 +248,7 @@ function convert(::Type{CategoricalArray{T, N, R}}, A::AbstractArray{S, N}) wher
     # if order is defined for level type, automatically apply it
     L = leveltype(res)
     if hasmethod(isless, Tuple{L, L})
-        levels!(res.pool, sort(levels(res.pool)))
+        levels!(res, sort(levels(res)))
```

Comment:
As most of the time `L` will be sortable maybe it is more efficient to first define levels for `res` in a correct order to avoid reordering as I understand now it is expensive?

Reply:
Yes but we don't know the levels in advance, and it will probably be slower to call `unique`.

It would make sense to add a `levels` keyword argument at some point for cases where you know them.

Reply:
Right - for some reason I thought `unique` would be faster than recode, but indeed it should be slower.